### PR TITLE
Migrate ES6

### PR DESCRIPTION
**Pull request by osukaa**
Finally had the time to update the module. I'm just not sure how to enable and disable a linting rule for certain line of code ==> https://github.com/hapijs/tv/compare/master...osukaa:migrateNode4?expand=1#diff-858d9b3c8622623b0da43bd2ed68d89dR47

### CODE REVIEW

**Comment by osukaa** [comment]
Referencing #129 Still more work to do, but hopefully this weekend :)

File: source/js/app.js [hunk 1]
```diff
@@ -36,25 +37,27 @@ exports = module.exports = window.app = internals.App = {
...
+        const settingsStore = opts.settingsStore;
+        const protocol = window.location.protocol === 'https:' ? 'wss://' : 'ws://';
 
-        var ws = new WebSocket(protocol + host + ':' + port);
-        var webSocketManager = opts.webSocketManagerClass.create(ws);
+        const ws = new WebSocket(protocol + host + ':' + port);
+        const webSocketManager = opts.webSocketManagerClass.create(ws);
 
-        var appView = new opts.appViewClass({
+        /* $lab:coverage:off$ */
```
Comment (Marsup) [inline source/js/app.js]: Why switch this part off ? It should be covered if others are.

File: source/js/messageParser.js [hunk 2]
```diff
@@ -158,8 +159,8 @@ internals.MessageParser.prototype._refreshResponseTimeout = function (message) {
     }
 
     if (!this._isResponse(message)) {
-        var self = this;
-        request.timer = setTimeout(function (){
+        const self = this;
```
Comment (Marsup) [inline source/js/messageParser.js]: The purpose of the => is to avoid the closure.

File: source/js/models/request.js [hunk 3]
```diff
@@ -1,13 +1,14 @@
+'use strict';
 // Load modules
 
-var Backbone = require('backbone');
+const Backbone = require('backbone');
 
-var DateTimeFormatter = require('../utils/dateTimeFormatter');
+// const DateTimeFormatter = require('../utils/dateTimeFormatter');
```
Comment (Marsup) [inline source/js/models/request.js]: Shouldn't it be removed instead ?

File: source/js/models/settings.js [hunk 4]
```diff
@@ -26,17 +27,17 @@ exports = module.exports = internals.Settings = Backbone.Model.extend({
...
+        this.on('change:channel', (model, channel) => {
 
             options.webSocketManager.applyFilter(channel);
         });
     },
 
     _updateSettingsStore: function (model) {
 
-        var self = this;
+        const self = this;
```
Comment (Marsup) [inline source/js/models/settings.js]: No need for the self here either.

File: source/js/utils/clientIdGenerator.js [hunk 5]
```diff
@@ -1,21 +1,22 @@
+'use strict';
 // Load modules
-var _ = require('lodash');
-var Chance = require('chance');
+// const _ = require('lodash');
```
Comment (Marsup) [inline source/js/utils/clientIdGenerator.js]: Remove ?

File: source/js/utils/search/searchCriterion.js [hunk 6]
```diff
@@ -120,9 +121,9 @@ internals.SearchCriterion.prototype._parseScopedProperty = function () {
 
 internals.SearchCriterion.prototype._matchesScopedProperty = function (request) {
 
-    var self = this;
+    const self = this;
```
Comment (Marsup) [inline source/js/utils/search/searchCriterion.js]: Not needed either.

File: source/js/views/channelSelector.js [hunk 7]
```diff
@@ -1,13 +1,14 @@
+'use strict';
 // Load modules
 
-var Backbone = require('backbone');
+const Backbone = require('backbone');
 
-var SettingsStore = require('../settingsStore');
+// const SettingsStore = require('../settingsStore');
```
Comment (Marsup) [inline source/js/views/channelSelector.js]: Remove ?

File: source/js/views/feedBody.js [hunk 8]
```diff
@@ -92,17 +93,17 @@ exports = module.exports = internals.FeedBodyView = Backbone.View.extend({
 
     _addRequest: function (request) {
 
-        var requestView = new RequestView({ model: request }).render();
+        const requestView = new RequestView({ model: request }).render();
         this._requestViews.push(requestView);
 
-        var self = this;
+        const self = this;
```
Comment (Marsup) [inline source/js/views/feedBody.js]: Not needed either.

File: source/js/views/serverLogs.js [hunk 9]
```diff
@@ -1,15 +1,16 @@
+'use strict';
 // Load modules
 
-var Backbone = require('backbone');
-var ZeroClipboard = require('zeroclipboard');
-var _ = require('lodash');
+const Backbone = require('backbone');
+const ZeroClipboard = require('zeroclipboard');
+// const _ = require('lodash');
```
Comment (Marsup) [inline source/js/views/serverLogs.js]: Remove ?

File: source/js/views/serverLogs.js [hunk 10]
```diff
@@ -22,10 +23,10 @@ exports = module.exports = internals.ServerLogsView = Backbone.View.extend({
 
     initialize: function () {
 
-        var self = this;
-        this.listenTo(this.collection, 'add', function (model) {
+        const self = this;
+        this.listenTo(this.collection, 'add', (model) => {
 
-            this.render();
+            self.this.render();
```
Comment (Marsup) [inline source/js/views/serverLogs.js]: Probably a typo here ? You actually need to keep it a normal function if you need its `this`.

File: source/js/views/serverLogs.js [hunk 11]
```diff
@@ -40,12 +41,12 @@ exports = module.exports = internals.ServerLogsView = Backbone.View.extend({
 
     _initializeClipboard: function () {
 
-        clipboard = this._createZeroClipboard();
+        const clipboard = this._createZeroClipboard();
 
-        var self = this;
-        clipboard.on('ready', function (readyEvent) {
+        const self = this;
```
Comment (Marsup) [inline source/js/views/serverLogs.js]: Not needed either.

File: source/js/views/settings.js [hunk 12]
```diff
@@ -1,15 +1,16 @@
...
 // Load modules
 
-var _ = require('lodash');
-var Backbone = require('backbone');
+const _ = require('lodash');
+const Backbone = require('backbone');
 
-var SettingsStore = require('../settingsStore');
-var JQuerySnippetGenerator = require('../utils/jQuerySnippetGenerator');
+// const SettingsStore = require('../settingsStore');
```
Comment (Marsup) [inline source/js/views/settings.js]: Remove ?

File: source/js/webSocketManager.js [hunk 13]
```diff
@@ -1,18 +1,19 @@
+'use strict';
 // Load modules
 
-var _ = require('lodash');
+// const _ = require('lodash');
```
Comment (Marsup) [inline source/js/webSocketManager.js]: Remove ?

File: source/js/webSocketManager.js [hunk 14]
```diff
@@ -1,18 +1,19 @@
...
-var internals = {};
+const internals = {};
 
 
 exports = module.exports = internals.WebSocketManager = function (webSocket) {
 
     this._webSocket = webSocket;
 
-    var self = this;
+    const self = this;
```
Comment (Marsup) [inline source/js/webSocketManager.js]: Not needed either.

File: test/client/app.js [hunk 15]
```diff
@@ -1,27 +1,28 @@
...
 
 describe('App', function () {
 
     describe('#start', function () {
 
-        beforeEach(function (){
+        beforeEach(function () {
 
-            var self = this;
+            const self = this;
```
Comment (Marsup) [inline test/client/app.js]: Could be removed if the function down there was =>.

File: test/client/views/request.js [hunk 16]
```diff
@@ -131,7 +132,7 @@ describe('RequestView', function () {
 
                 this.$favorite = this.view.$('.favorite');
 
-                var self = this;
+                const self = this;
```
Comment (Marsup) [inline test/client/views/request.js]: Not needed either.

**Comment by Marsup** [comment]
You did the most tedious part, good luck making it work now :)
You could have used the implicit return of the => a bit more, but that's just a stylistic preference.

**Comment by osukaa** [comment]
@Marsup Yeah! It was really tedious :s  hopefully it'll get easier to fix all the thing you pointed out! and get this completely updated.

**Comment by gergoerdosi** [comment]
Any news on this PR?

**Comment by osukaa** [comment]
@gergoerdosi I need help with the PhantomJS stuff :/ the comments made by Marsup I already fixed them.

**Comment by gergoerdosi** [comment]
What's the issue exactly with PhatomJS? I wanted to look into this, but tests run fine locally:

```
Gergo:tv Gergo$ node -v
v4.4.4
Gergo:tv Gergo$ npm test

> tv@5.0.2 test /hapijs/tv
> mocha-phantomjs test/client/index.html --hooks test/client/phantomHooks.js && lab test/index.js -a code -t 100 -L



  SearchCriteria
    .create
      ✓ returns a new instance of SearchCriteria
      ✓ creates an array of SearchCriterion objects
    #matches
      with all criterion that match the request
        ✓ returns true
      with a ignored search criterion
        ✓ skips that criterion during evaluation
      with at least one criterion that doesn't match the request
        ✓ returns false

  SearchCriterion
    .create
      ✓ returns a new instance of SearchCriterion
      with a scoped search criterion
        ✓ marks the criterion as scoped
        ✓ specifies the scoped property
      with a scoped search criterion that doesn't specify a value
        ✓ marks the criterion as ignored
      with a general search criterion that contains a colon
        ✓ does not ignore search criterion or mark it as ignored
      with a general search criterion
        ✓ marks the criterion as not scoped
        ✓ does not set a scoped property
        ✓ marks the criterion as not ignored
    #matches
      with a scoped criterion
        with a single value
          that matches the request
            ✓ returns true
          that doesn't match the request
            ✓ returns false
        with multiple values
          with one that matches the request
            ✓ returns true
          with none that match the request
            ✓ returns false
          and a trailing comma
            ✓ ignores the empty value
      with a general criterion
        that matches the request
          ✓ returns true
        that doesn't match the request
          ✓ returns false
      without a status code
        ✓ doesn't error

  ClientIdGenerator
    #generate
      ✓ returns a code
      ✓ always returns a 'unique' code

  DateTimeFormatter
    #longTime
      ✓ returns time in a long format
    #shortDate
      ✓ returns date in a short format

  HandlebarsHelpers
    ✓ includes DateTimeFormatter functions as Helpers
    ✓ registers helper functions with handlerbars
    jsonMarkup
      ✓ transforms json to an html representation
    jQuerySnippet
      ✓ generates a jquery snippet
    isEq
      with matching arguments
        ✓ calls the main function
      without matching arguments
        ✓ calls the inverse function
    tagColor
      with a supported color tag
        ✓ returns the tag name to be used as the css class
      without a supported color tag
        ✓ doesn't return anything

  jQuerySnippetGenerator
    .generate
      ✓ generates a snippet with the given clientId

  RequestToTextConverter
    #convertToText
      ✓ converts the request to text

  Request
    #hasError
      ✓ returns true if request timed out
      ✓ returns true if status code is 500
      ✓ returns false otherwise
    #hasWarning
      ✓ returns true if statusCode inbetween 400 and 500
      ✓ returns false otherwise
    #toJSON
      ✓ parses all subcomponents

  Settings
    #defaults
      ✓ sets the expected defaults from settings
    #initialize
      ✓ applies the filter to the websocket when channel is changed
      ✓ updates the store when properties change with the properties that changed

  AppView
    #render
      ✓ renders the toolbar
      ✓ renders the settings modal (hidden)
      ✓ renders the feed header
      ✓ renders the feed body
      when visiting the app for the first time
        - defaults the channel to "all"
        - defaults the client id to a random generated value
        - shows the settings view
      when the toolbar triggers
        "search criteria changed"
          - filters the body's requests by the search criteria
        "show settings"
          - shows the settings view
        clear feed
          - "clears the feed body"
      when the feed header triggers
        "toggle favorites"
          - toggles favorites on the feed body
        "collapse all"
          - collapses all the requests in the feed body
      when the feed body triggers
        "request expand toogle"
          with a request collapse and no other requests expanded
            - disables the feed header's collapse all action
          with a request expanded
            - enables the feed header's collapse all action
        "request favorite toogle"
          with a request unfavorited and no other requests favorited
            - disables the feed header's favorites filter action
          with a request favorited
            - enables the feed header's favorites filter action

  ChannelSelectorView
    #template
      ✓ returns html
    #events
      with a click on a button
        ✓ sets the settings model channel property
    #initialize
      with a change to the settings model clientId
        ✓ updates the clientId button html
      with a change to the settings model channel
        ✓ applies the active class to only the button for the new channel
    #render
      ✓ has the expected markup

  FeedBody
    when a request is add to the collection
      - adds a request to the feed body
    #render
      - clears the body
    #clear
      - clears out the feed body's markup
      - removes the request views
    #hasFavoritedRequests
      with a least one request that's favorited
        - returns true
      without any requests favorited
        - returns false
    #hasExpandedRequests
      with a least one request that's expanded
        - returns true
      without any requests expanded
        - returns false
    #collapseAll
      - puts each request into a collapsed state
    with favorites currently not filtered
      #toggleFavorites
        - hides all non-favorited requests
    with favorites currently filtered
      #toggleFavorites
        - shows all non-favorited requests
    #filterRequests
      with a present search criteria query string
        - it hides any requests that don't match the search criteria filter
      with an empty search criteria query string
        - it shows any requests that were hidden by the previous search criteria filter
    when a request is collapsed/expanded
      - triggers requestExpandToggle
    when a request is favorited/unfavorited
      - triggers requestFavoriteToggle
    when a request status changes
      with a search criteria filter set to display that status
        - the request is shown
      with a search criteria filter set to hide that status
        - the request is hidden
    with the browser window scrolled all the way to the bottom
      when a new request is added to the feed's body
        - scrolls the screen back to the bottom

  FeedHeader
    when the collapse all action is clicked
      - all expanded requests get collapsed
    with the favorites filter turned off
      when the favorites filter action is clicked
        - displays the favorites filter as turned on
        - triggers "toggleFavorites" with the toggle set to true
    with the favorites filter turned on
      when the favorites filter action is clicked
        - displays the favorites filter as turned off
        - triggers "toggleFavorites" with the toggle set to false
    #clear
      - disables the favorites filter action
      - disables the collapse all action
    #enableCollapseAll
      - enables the collapse all action
    #disableCollapseAll
      - disables the collapse all action
    #enableFavoritesFilter
      - enables the favorited filter action
    #disableFavoritesFilter
      - disables the favorited filter action

  RequestView
    #template
      ✓ returns html
    #events
      with a click on .request-details
        ✓ toggles the server logs pane
        ✓ builds the server logs view once and only once
        ✓ toggles .active
        ✓ emits the "serverLogsToggle" event with "true" when shown
        ✓ emits the "serverLogsToggle" event with "false" when hidden
      with a click on .favorite
        ✓ toggles .favorited
        ✓ toggles the .active class
        ✓ emits the "favoriteToggle" event with "true" when favorited
        ✓ emits the "favoriteToggle" event with "false" when unfavorited
    #initialize
      with a change to the settings model clientId
        ✓ sets default state
    #render
      ✓ has the expected markup

  RequestDetailsView
    when the model changes
      - re-renders
    #render
      with a request that has an error
        - displays that the request has an error
      with a request that has a warning
        - displays that the request has a warning

  ServerLogsView
    .template
      ✓ returns html
    #initialize
      ✓ rerenders when a new model is added to the collection
    .events
      with a click on .data
        ✓ toggles the expanded class on the server log row
    #render
      ✓ returns the expected number of server log rows
      ✓ returns the expected tags

  SettingsView
    when the client id changes
      - updates the jquery snippet
    when submit is clicked
      - sets the client id on the model
      - hides itself
      with a changed client id
        - sets the client id as the channel on the model
    when cancel is clicked
      - resets the views client id back to original client id
      - hides itself
      - re-renders itself to reset it's display for the next time it's shown
    when the original settings model is changed
      - changes the view's model
    #show
      - shows the view
    #hide
      - hides the view

  ToolbarView
    when search criteria changes
      - triggers "searchChanged", passing through the search criteria
    when settings is clicked
      - triggers "showSettings"
    when clear is clicked
      - triggers "clearFeed"
    when pause is clicked
      - it triggers "pause"
    when resume is clicked
      - it triggers "resume"

  App
    #start
      ✓ renders the App view
      when the socket is openned
        ✓ sets the channel as the web socket's filter
      when a message is received
        ✓ adds the message to the message parser

  MessageParser
    #create
      ✓ creates a new instance of a MessageParser
    #addMessage
      with an initial message for a request
        ✓ creates a new request
        ✓ creates a new server log
        with an empty response message
          ✓ does not add the message to the response's server logs
        with a subsequent message for a request
          ✓ doesn't create a request
          ✓ creates a new server log
        with a response message for a request
          ✓ updates the request object with the status code
          ✓ adds a response tag to the server log message
        with a response message for a request
          ✓ updates the request object with the status code
          ✓ adds a response tag to the server log message
      with a non "received" message for a request that doesn't exist
        ✓ does not create a request
      when a response doesn't come in within a set timeout
        ✓ marks the request as having a response timeout
        ✓ calls the onResponseTimeout callback
        ✓ overrides the timeout if a subsequent received message comes in after the timeout
      when a server log for a requests comes in before the response timeout
        ✓ resets the response timeout for that request
      when a response for a request comes in after the response timeout has occured
        ✓ clears the response error timeout
      when a response for a request comes in after the response timeout has occured
        ✓ clears the response error timeout

  SettingsStore
    #exists
      ✓ returns true if the value is in localStorage
      ✓ returns false if the value is in localStorage
    #get
      ✓ returns the value if found
    #set
      ✓ sets the value in localStorage

  WebSocketManager
    #create
      ✓ creates a new instance of a WebSocketManager
    when a websocket is openned
      ✓ calls teh onSocketOpen callback
    #applyFilter
      with a webSocket connection that is open
        ✓ filters the messages by the given client id
    #clearFilter
      ✓ clears any previously set filter so all messages are received
    #onMessage
      ✓ registers a callback that's called whenever a message is received
    #pause
      ✓ stops forwarding messages to the onMesssage callback
    #resume
      ✓ resumes forwarding messages to the onMesssage callback


  101 passing (456ms)
  59 pending

Writing coverage to coverage/coverage.json

  
  .........

9 tests complete
Test duration: 1011 ms
Assertions count: 38 (verbosity: 4.22)
No global variable leaks detected
Coverage: 100.00%
Linting results: No issues
```

```
Gergo:tv Gergo$ node -v
v6.2.0
Gergo:tv Gergo$ npm test

> tv@5.0.2 test /hapijs/tv
> mocha-phantomjs test/client/index.html --hooks test/client/phantomHooks.js && lab test/index.js -a code -t 100 -L



  SearchCriteria
    .create
      ✓ returns a new instance of SearchCriteria
      ✓ creates an array of SearchCriterion objects
    #matches
      with all criterion that match the request
        ✓ returns true
      with a ignored search criterion
        ✓ skips that criterion during evaluation
      with at least one criterion that doesn't match the request
        ✓ returns false

  SearchCriterion
    .create
      ✓ returns a new instance of SearchCriterion
      with a scoped search criterion
        ✓ marks the criterion as scoped
        ✓ specifies the scoped property
      with a scoped search criterion that doesn't specify a value
        ✓ marks the criterion as ignored
      with a general search criterion that contains a colon
        ✓ does not ignore search criterion or mark it as ignored
      with a general search criterion
        ✓ marks the criterion as not scoped
        ✓ does not set a scoped property
        ✓ marks the criterion as not ignored
    #matches
      with a scoped criterion
        with a single value
          that matches the request
            ✓ returns true
          that doesn't match the request
            ✓ returns false
        with multiple values
          with one that matches the request
            ✓ returns true
          with none that match the request
            ✓ returns false
          and a trailing comma
            ✓ ignores the empty value
      with a general criterion
        that matches the request
          ✓ returns true
        that doesn't match the request
          ✓ returns false
      without a status code
        ✓ doesn't error

  ClientIdGenerator
    #generate
      ✓ returns a code
      ✓ always returns a 'unique' code

  DateTimeFormatter
    #longTime
      ✓ returns time in a long format
    #shortDate
      ✓ returns date in a short format

  HandlebarsHelpers
    ✓ includes DateTimeFormatter functions as Helpers
    ✓ registers helper functions with handlerbars
    jsonMarkup
      ✓ transforms json to an html representation
    jQuerySnippet
      ✓ generates a jquery snippet
    isEq
      with matching arguments
        ✓ calls the main function
      without matching arguments
        ✓ calls the inverse function
    tagColor
      with a supported color tag
        ✓ returns the tag name to be used as the css class
      without a supported color tag
        ✓ doesn't return anything

  jQuerySnippetGenerator
    .generate
      ✓ generates a snippet with the given clientId

  RequestToTextConverter
    #convertToText
      ✓ converts the request to text

  Request
    #hasError
      ✓ returns true if request timed out
      ✓ returns true if status code is 500
      ✓ returns false otherwise
    #hasWarning
      ✓ returns true if statusCode inbetween 400 and 500
      ✓ returns false otherwise
    #toJSON
      ✓ parses all subcomponents

  Settings
    #defaults
      ✓ sets the expected defaults from settings
    #initialize
      ✓ applies the filter to the websocket when channel is changed
      ✓ updates the store when properties change with the properties that changed

  AppView
    #render
      ✓ renders the toolbar
      ✓ renders the settings modal (hidden)
      ✓ renders the feed header
      ✓ renders the feed body
      when visiting the app for the first time
        - defaults the channel to "all"
        - defaults the client id to a random generated value
        - shows the settings view
      when the toolbar triggers
        "search criteria changed"
          - filters the body's requests by the search criteria
        "show settings"
          - shows the settings view
        clear feed
          - "clears the feed body"
      when the feed header triggers
        "toggle favorites"
          - toggles favorites on the feed body
        "collapse all"
          - collapses all the requests in the feed body
      when the feed body triggers
        "request expand toogle"
          with a request collapse and no other requests expanded
            - disables the feed header's collapse all action
          with a request expanded
            - enables the feed header's collapse all action
        "request favorite toogle"
          with a request unfavorited and no other requests favorited
            - disables the feed header's favorites filter action
          with a request favorited
            - enables the feed header's favorites filter action

  ChannelSelectorView
    #template
      ✓ returns html
    #events
      with a click on a button
        ✓ sets the settings model channel property
    #initialize
      with a change to the settings model clientId
        ✓ updates the clientId button html
      with a change to the settings model channel
        ✓ applies the active class to only the button for the new channel
    #render
      ✓ has the expected markup

  FeedBody
    when a request is add to the collection
      - adds a request to the feed body
    #render
      - clears the body
    #clear
      - clears out the feed body's markup
      - removes the request views
    #hasFavoritedRequests
      with a least one request that's favorited
        - returns true
      without any requests favorited
        - returns false
    #hasExpandedRequests
      with a least one request that's expanded
        - returns true
      without any requests expanded
        - returns false
    #collapseAll
      - puts each request into a collapsed state
    with favorites currently not filtered
      #toggleFavorites
        - hides all non-favorited requests
    with favorites currently filtered
      #toggleFavorites
        - shows all non-favorited requests
    #filterRequests
      with a present search criteria query string
        - it hides any requests that don't match the search criteria filter
      with an empty search criteria query string
        - it shows any requests that were hidden by the previous search criteria filter
    when a request is collapsed/expanded
      - triggers requestExpandToggle
    when a request is favorited/unfavorited
      - triggers requestFavoriteToggle
    when a request status changes
      with a search criteria filter set to display that status
        - the request is shown
      with a search criteria filter set to hide that status
        - the request is hidden
    with the browser window scrolled all the way to the bottom
      when a new request is added to the feed's body
        - scrolls the screen back to the bottom

  FeedHeader
    when the collapse all action is clicked
      - all expanded requests get collapsed
    with the favorites filter turned off
      when the favorites filter action is clicked
        - displays the favorites filter as turned on
        - triggers "toggleFavorites" with the toggle set to true
    with the favorites filter turned on
      when the favorites filter action is clicked
        - displays the favorites filter as turned off
        - triggers "toggleFavorites" with the toggle set to false
    #clear
      - disables the favorites filter action
      - disables the collapse all action
    #enableCollapseAll
      - enables the collapse all action
    #disableCollapseAll
      - disables the collapse all action
    #enableFavoritesFilter
      - enables the favorited filter action
    #disableFavoritesFilter
      - disables the favorited filter action

  RequestView
    #template
      ✓ returns html
    #events
      with a click on .request-details
        ✓ toggles the server logs pane
        ✓ builds the server logs view once and only once
        ✓ toggles .active
        ✓ emits the "serverLogsToggle" event with "true" when shown
        ✓ emits the "serverLogsToggle" event with "false" when hidden
      with a click on .favorite
        ✓ toggles .favorited
        ✓ toggles the .active class
        ✓ emits the "favoriteToggle" event with "true" when favorited
        ✓ emits the "favoriteToggle" event with "false" when unfavorited
    #initialize
      with a change to the settings model clientId
        ✓ sets default state
    #render
      ✓ has the expected markup

  RequestDetailsView
    when the model changes
      - re-renders
    #render
      with a request that has an error
        - displays that the request has an error
      with a request that has a warning
        - displays that the request has a warning

  ServerLogsView
    .template
      ✓ returns html
    #initialize
      ✓ rerenders when a new model is added to the collection
    .events
      with a click on .data
        ✓ toggles the expanded class on the server log row
    #render
      ✓ returns the expected number of server log rows
      ✓ returns the expected tags

  SettingsView
    when the client id changes
      - updates the jquery snippet
    when submit is clicked
      - sets the client id on the model
      - hides itself
      with a changed client id
        - sets the client id as the channel on the model
    when cancel is clicked
      - resets the views client id back to original client id
      - hides itself
      - re-renders itself to reset it's display for the next time it's shown
    when the original settings model is changed
      - changes the view's model
    #show
      - shows the view
    #hide
      - hides the view

  ToolbarView
    when search criteria changes
      - triggers "searchChanged", passing through the search criteria
    when settings is clicked
      - triggers "showSettings"
    when clear is clicked
      - triggers "clearFeed"
    when pause is clicked
      - it triggers "pause"
    when resume is clicked
      - it triggers "resume"

  App
    #start
      ✓ renders the App view
      when the socket is openned
        ✓ sets the channel as the web socket's filter
      when a message is received
        ✓ adds the message to the message parser

  MessageParser
    #create
      ✓ creates a new instance of a MessageParser
    #addMessage
      with an initial message for a request
        ✓ creates a new request
        ✓ creates a new server log
        with an empty response message
          ✓ does not add the message to the response's server logs
        with a subsequent message for a request
          ✓ doesn't create a request
          ✓ creates a new server log
        with a response message for a request
          ✓ updates the request object with the status code
          ✓ adds a response tag to the server log message
        with a response message for a request
          ✓ updates the request object with the status code
          ✓ adds a response tag to the server log message
      with a non "received" message for a request that doesn't exist
        ✓ does not create a request
      when a response doesn't come in within a set timeout
        ✓ marks the request as having a response timeout
        ✓ calls the onResponseTimeout callback
        ✓ overrides the timeout if a subsequent received message comes in after the timeout
      when a server log for a requests comes in before the response timeout
        ✓ resets the response timeout for that request
      when a response for a request comes in after the response timeout has occured
        ✓ clears the response error timeout
      when a response for a request comes in after the response timeout has occured
        ✓ clears the response error timeout

  SettingsStore
    #exists
      ✓ returns true if the value is in localStorage
      ✓ returns false if the value is in localStorage
    #get
      ✓ returns the value if found
    #set
      ✓ sets the value in localStorage

  WebSocketManager
    #create
      ✓ creates a new instance of a WebSocketManager
    when a websocket is openned
      ✓ calls teh onSocketOpen callback
    #applyFilter
      with a webSocket connection that is open
        ✓ filters the messages by the given client id
    #clearFilter
      ✓ clears any previously set filter so all messages are received
    #onMessage
      ✓ registers a callback that's called whenever a message is received
    #pause
      ✓ stops forwarding messages to the onMesssage callback
    #resume
      ✓ resumes forwarding messages to the onMesssage callback


  101 passing (364ms)
  59 pending

Writing coverage to coverage/coverage.json

  
  .........(node:36803) DeprecationWarning: 'root' is deprecated, use 'global'


9 tests complete
Test duration: 995 ms
Assertions count: 38 (verbosity: 4.22)
The following leaks were detected:Reflect
Coverage: 100.00%
Linting results: No issues

npm ERR! Test failed.  See above for more details.
```

**Comment by osukaa** [comment]
Did you tested my branch locally? I was having a problem of an array being bigger than it was expecting it on the test.

**Comment by gergoerdosi** [comment]
No, sorry. I fetched your branch, but forgot to switch to it. I get locally what is on Travis:

```
$ npm test

> tv@5.0.2 test /Users/Gergo/Workspace/hapijs/tv
> mocha-phantomjs -p node_modules/phantomjs-prebuilt/lib/phantom/bin/phantomjs test/client/index.html --hooks test/client/phantomHooks.js && lab test/index.js -a code -t 100 -L

mocha.run() was called with no tests
npm ERR! Test failed.  See above for more details.
```

**Comment by jonathansamines** [comment]
Hi @osukaa I just PR your fork to propose a solution to the phantom issues.

**Comment by Marsup** [comment]
Brought it back on this repo on https://github.com/hapijs/tv/tree/es6 with most of your changes @jonathansamines, this shouldn't be on a fork. Tests are still failing though, I'll let you all sort it out.

**Comment by jonathansamines** [comment]
@osukaa @Marsup My changes at:

- [settings.js](https://github.com/osukaa/tv/pull/1/files#diff-d71dcbed8b810a691c0f2ccb74d3faca)
- [websocketmanager.js](https://github.com/osukaa/tv/pull/1/files#diff-00e587439cb9d84cc4aa5f30fc9d3251)

Are still needed in order to "this" refer to the right object, then therefore make the test-suit pass :)

**Comment by Marsup** [comment]
Sorry, I got confused by working on @osukaa's fork, 3-way merge gone wrong.

**Comment by Marsup** [comment]
@jonathansamines updated the branch and put you as author if you don't mind.

**Comment by jonathansamines** [comment]
@Marsup @osukaa great!, should the `es6` branch replace this PR? Is that something I can do, or is @osukaa or you doing that?

**Pull request closed without merging**